src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/OperatorApiServiceImpl.scala

```diff
@@ -38,19 +37,19 @@ final case class OperatorApiServiceImpl(
 
     val result: Future[ClientKeys] = for {
       clientUuid    <- clientId.toFutureUUID
-      client        <- authorizationManagementService.getClient(clientUuid)(contexts)
+      client        <- authorizationManagementService.getClient(clientUuid)
       _             <- assertIsClientConsumer(client).toFuture
       operatorUuid  <- operatorId.toFutureUUID
       relationships <- partyManagementService.getRelationshipsByPersonId(operatorUuid, Seq.empty)
       clientUuid    <- clientId.toFutureUUID
-      clientKeys    <- authorizationManagementService.getClientKeys(clientUuid)(contexts)
-      operatorKeys = clientKeys.keys.filter(key => relationships.items.exists(_.id == key.relationshipId))
-      keysResponse = authorizationmanagement.client.model.KeysResponse(operatorKeys)
-    } yield ClientKeys(keysResponse.keys.map(_.toApi))
+      clientKeys    <- authorizationManagementService.getClientKeys(clientUuid)
+      keys <- Future.traverse(clientKeys.filter(key => relationships.items.exists(_.id == key.relationshipId)))(f =>
+        f.toClientKeyApi.toFuture
+      )
```

Comment:
No need to use threads in this case
```suggestion
      keys <- clientKeys.filter(key => relationships.items.exists(_.id == key.relationshipId)).traverse(_.toClientKeyApi
      ).toFuture
```